internal/postrs/api_test.go

```diff
@@ -0,0 +1,179 @@
+package postrs
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/spacemeshos/post/internal/gpu"
+)
+
+var (
+	commitment        = make([]byte, 32)
+	defaultDifficulty = new(big.Int).Sub(new(big.Int).Lsh(big.NewInt(1), 256), big.NewInt(1)).Bytes()
+)
+
+func TestCPUProviderExists(t *testing.T) {
+	id := CPUProviderID()
+	require.NotZero(t, id, "CPU provider not found")
+
+	providers, err := OpenCLProviders()
+	require.NoError(t, err)
+
+	for _, p := range providers {
+		if p.ID == id {
+			require.NotEmpty(t, p.Model)
+			require.Equal(t, ClassCPU, p.DeviceType)
+			return
+		}
+	}
+
+	require.Fail(t, "CPU provider doesn't exist")
+}
+
+// TestScryptPositions is an output correctness sanity test. It doesn't cover many cases.
+func TestScryptPositions(t *testing.T) {
+	providers, err := OpenCLProviders()
+	require.NoError(t, err)
+
+	vrfDifficulty := make([]byte, 32)
+	copy(vrfDifficulty, defaultDifficulty)
+	vrfDifficulty[0] = 0
+
+	start := uint64(1)
+	end := uint64(1 << 8)
```

Comment:
It carried over from the previous implementation and I personally have no issues with thinking in powers of two. I assume it might have been som performance thing where `gpu-post` was initializing in batches that were a power of 2 in size?

I don't have strong feelings either way, I just find with larger numbers, numbers of the kind `uint64(1 << 32)` become easier to parse visually and compare to each other than numbers of the kind `uint64(4000000000)`.

Do you want me to change the tests?